src/data/ast.ts

```diff
@@ -1,267 +1,98 @@
-
-        // first we get the remaining part of the FROM phrase. ") as r"
-        const fromPhrase = ast.get('FROM');
-        const fromPhraseAfterTableName = fromPhrase!
-          [fromPhrase!.length - 1]!.toString()
-          .trim()
-          .substring(targetTable.length);
-
-        // if the fromPhraseAfterTheTableName is not the ending of a bracket statement, then don't add it
-        if (!fromPhraseAfterTableName.includes(')')) {
-          ast.set('WHERE', [whereClause]);
-          continue;
-        }
-        // apply the remaining part of the FROM phrase to the end of the new WHERE clause
-        ast.set('WHERE', [`${whereClause} ${fromPhraseAfterTableName}`]);
-        // set the FROM clause to only have the table name
-        const index = ast.get('FROM')!.indexOf(clause);
-        ast.get('FROM')![index] = ` ${targetTable} `;
-        continue;
+      const rft = replaceFuncs.find((x) => x.replacementName === t.name);
+      if (rft) {
+        return { ...t, name: t.name.replace(rft.replacementName, rft.name) };
       }
-      where!.unshift(`${whereClause} AND `);
-    }
-  }
-
-  // Each node in the AST needs to be checked to see if ad hoc filters should be applied
-  ast.forEach((clauses: Clause[]) => {
-    for (const c of clauses) {
-      if (c !== null && !isString(c)) {
-        applyFiltersToAST(c, whereClause, targetTable);
+      return map.super().tableRef(t);
+    },
+    ref: (r) => {
+      const rf = replaceFuncs.find((x) => r.name.startsWith(x.replacementName));
+      if (rf) {
+        const d = r.name.replace(rf.replacementName, rf.name);
+        return { ...r, name: d };
       }
-    }
-  });
-
-  return ast;
+      return map.super().ref(r);
+    },
+    call: (c) => {
+      const rf = replaceFuncs.find((x) => c.function.name.startsWith(x.replacementName));
+      if (rf) {
+        return { ...c, function: { ...c.function, name: c.function.name.replace(rf.replacementName, rf.name) } };
+      }
+      return map.super().call(c);
+    },
+  }));
+  return mapper.statement(ast)!;
 }
 
-export function removeConditionalAllsFromAST(ast: AST, queryVarNames: string[]): AST {
-  if (!ast || !ast.get('FROM')) {
-    return ast;
-  }
-
-  const where = ast.get('WHERE');
-  if (where) {
-    for (let i = 0; i < where.length; i++) {
-      const c = where[i];
-      if (isString(c) && queryVarNames.some((v) => c.includes(v)) && c.includes('$')) {
-        // remove AND/OR before this condition if this is the last condition
-        if (i === where.length - 1) {
-          where.splice(i - 1, 2);
-        }
-        // remove AND/OR after this condition
-        if (where.length > 1) {
-          where.splice(i, 2);
-          i--;
-        }
-        // moves the ending of the phrase, like ')', to the next logical place
-        movePhraseEnding(c, ast);
+export function getTable(sql: string): string {
+  const stm = sqlToStatement(sql) as SelectFromStatement;
```

Comment:
Rather than an `as` here you can check the value of `stm.type`. If it === `select` then the type will be correctly inferred as `SelectFromStatement`. You can then remove this as and the `as FromTable` and `as FromStatement`

Reply:
all these `as` statements! my c# is showing